### ISSUE TITLE
Race mode

### ISSUE THREAD

**Issue by Fransje** [inexor-game/code#105]
The huge popularity of the race servers makes me think this should be a builtin mode. Also attracts a wider audience: players who like parkour more than shooting might stick to Inexor because of this.
There would also have to be a number of professional race maps in the release.

**Comment by Croydon** [comment]
The predecessor of Inexor ("Sauerbomber") had already a race mode included, however it was decided to remove it because of a bad implementation. In my personal opinion it isn't a high priority nor absolutely necessary, but an idea to think about later.

**Comment by Fransje** [comment]
Ok, well an implementation like the race servers have now (Racing Reloaded for example) is perfect. I understand it's not high priority, but some more thinking about it in the future would be great :)

**Comment by IAmNotHanni** [comment]
Once visual scripting will be implemented I think you do not really game modes at all
you can script all the logic into the map itself...

**Comment by a-teammate** [comment]
@IAmNotHanni Uhm ok, i dont think that will be a good idea. (if you wanted to say: "no gamemode at all")

Gamemodes will still be necessary and it shouldnt be advised to get rid of those completely. It would complicate stuff.
You'd need to redefine stuff for gamemodes in every map. (Alternatively you could ofc tell the map to import those gamemode-definitions. But that would just make sense if you subsequently allow the map to change or extend those definitions. I'd have pretty strong concerns about those steps, since a main aspect of games is to offer an "overall look and feel" and playing on reissen an other instactf than on dust6 would totally oppose that)

@Fransje I think the racemode will find its way into inexors release somehow :)

**Comment by aschaeffer** [comment]
Whether built in or as a custom game mode, i think a race mode will come. So I wouldn't close the ticket right, especially without any comment. A race mode has no high priority, because we have to make the base for custom game modes first: the entity system (#78) and the flow graph.

**Comment by Fohlen** [comment]
Once we have the possibility to mod gamemodes (...) this topic may gain attention again, it should be easy to do in the future.
But I don't think it belongs into this bugtracker at all..
Maybe we should seperate gameplay specific requests onto another tracker :+1:

**Comment by Croydon** [comment]
I disagree with seperating gameplay requests in another issue tracker, that would be only an option to think about, when we would have a very hight abstract level of gameplay definitions (read: nothing hard coded).

**Comment by Fohlen** [comment]
Refers to #339
